gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java

```diff
@@ -320,12 +310,18 @@ public HttpJsonChannel getChannel() {
     return channel;
   }
 
+  public HttpJsonCallOptions getCallOptions() {
+    return callOptions;
+  }
+
+  @Deprecated
   public Instant getDeadline() {
-    return deadline;
+    return getCallOptions() != null ? getCallOptions().getDeadline() : null;
```

Comment:
I think it's good to mark it `@Nullable`.

Reply:
Marked `getDeadline()` and `getCredentials()` `@Nullable`